swift/example_code/s3/basics/Tests/basics-tests/S3DemoCleanup.swift

```diff
@@ -0,0 +1,225 @@
+/*
+   A class to record the Amazon S3 buckets and files created by a code example,
+   so that they can easily be cleaned up after the example is finished.
+
+   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0
+*/
+
+import Foundation
+import AWSS3
+import ClientRuntime
+import AWSClientRuntime
+import SwiftUtilities
+
+/// Errors thrown by the `S3DemoCleanup` class.
+enum S3DemoCleanupError: Error {
+    case duplicate          /// Attempted to create an item that already exists.
+    case notFound           /// Attempted to access an item that doesn't exist.
+}
+
+/// A structure identifying a file given its bucket name and the
+/// file's name within the bucket.
+public struct S3DemoFileInfo: Equatable {
+    /// The name of the bucket containing the file.
+    var bucket: String
+    /// The file's name.
+    var file: String
+    /// The contents of the file.
+    var contents: Data?
+
+    init(bucket: String, file: String, contents: Data? = nil) {
+        self.bucket = bucket
+        self.file = file
+        self.contents = contents
+    }
+
+    /// Determine if the `S3DemoFileInfo` matches the given bucket
+    /// name and file name.
+    ///
+    /// - Parameters:
+    ///   - bucket: Bucket name to compare to.
+    ///   - file: File name to compare to.
+    func matches(bucket: String, file: String) -> Bool {
+        return bucket == self.bucket && file == self.file
+    }
+}
+
+public class S3DemoCleanup {
+    /// Returns a single shared instance of `S3DemoCleanup` that should
+    /// always be used to access the cleanup manager.
+    var client: S3Client? = nil
+    var buckets: [String]
+    var files: [S3DemoFileInfo]
+
+    init() async {
+        do {
+            self.client = try await S3Client()
+        } catch {
+            print("Error initializing S3 client for test cleanup processor:")
+            dump(error)
+            exit(1)
+        }
+        self.buckets = []
+        self.files = []
+    }
+
+    /// Return the index into the tracked file list at which the specified
+    /// `S3DemoFileInfo` object is found.
+    ///
+    /// - Parameter fileInfo: The `S3DemoFileInfo` object to find in the tracked
+    ///   files list.
+    /// - Returns: The index into the tracked files list at which the file is
+    ///   located. If the file isn't in the list, the result is -1.
+    func getIndex(fileInfo: S3DemoFileInfo) -> Int {
+        var index = 0
+
+        for file in self.files {
+            if fileInfo.matches(bucket: file.bucket, file: file.file) {
+                return index
+            }
+            index += 1
+        }
+        return -1
+    }
+
+    /// Clean up all of the files and buckets previously added to the
+    /// demo cleanup manager.
```

Comment:
Is "demo cleanup manager" the same thing as the "cleanup manager" (line 50) and/or the "test cleanup processor" (line 59)?